scss/forms/_form-check.scss

```diff
@@ -107,6 +107,46 @@
 .form-switch {
   padding-left: $form-switch-padding-start;
 
+  @each $state, $value in $theme-colors {
+    &.switch-on-#{$state} {
+      .form-check-input {
+        &:checked {
+          background-color: $value;
+          border-color: $value;
+          &:focus,
+          &:active {
+            border-color: $value;
+            outline: 0;
+            box-shadow: 0 0 $input-btn-focus-blur $input-btn-focus-width rgba($value, $input-btn-focus-color-opacity);
+          }
+        }
+      }
+    }
```

Comment:
This could be refactored a bit, using custom properties and a generic class name.

```scss
.switch-on {
   .form-check-input {
        &:checked {
          background-color: var(--#{$variable-prefix}-switch-color);
          border-color: var(--#{$variable-prefix}-switch-color);
          &:focus,
          &:active {
            border-color: var(--#{$variable-prefix}-switch-color);
            outline: 0;
            box-shadow: 0 0 $input-btn-focus-blur $input-btn-focus-width rgba(var(--#{$variable-prefix}-switch-color-rgb), $input-btn-focus-color-opacity);
          }
        }
     }
}

@each $color, $value in $theme-colors {
    &.switch-on-#{$color} {
      --#{$variable-prefix}-switch-color: var(--#{$variable-prefix}#{$color});
      --#{$variable-prefix}-switch-color-rgb: var(--#{$variable-prefix}#{$color}-rgb);
    }
}
```

This should decrease the amount of compiled CSS.

Two things to notice:
1. the `switch-on` class is mandatory, so we'd need to use eg. `switch-on switch-on-primary`.
2. I renamed `$state` to `$color` since this variable does not describe a state but a color. It then allowed a simpler matching to existing theme colors' custom properties.

What do you think about this? Does it look reasonnable to you?

Reply:
BTW, if we go with this, I'd move the whole new CSS after the existing `.form-switch` code block:

```scss
.form-switch {
  … 
}

.switch-on {
  …
}

.switch-off {
  …
}
```

This would decrease specificity and compiled CSS output too.

Reply:
I'll try to help if needed, don't worry. Thanks a lot for your time!